src/main/java/cl/transbank/onepay/example/controller/TransactionController.java

```diff
@@ -25,8 +26,16 @@
 
     @RequestMapping(value = "/transaction-create", method = RequestMethod.POST)
     @ResponseBody
-    public String transactionCreate() throws AmountException {
+    public String transactionCreate(@RequestParam("channel") String channel) throws AmountException, UnknownHostException, SocketException {
+        String ip;
+        try(final DatagramSocket socket = new DatagramSocket()){
+            socket.connect(InetAddress.getByName("8.8.8.8"), 10002);
+            ip = socket.getLocalAddress().getHostAddress();
```

Comment:
What's the purpose of this `ip` variable and all that calculation? From what I can see below it's not used because the ip comes from the `HOST_IP` env variable.

Reply:
Yes that was my first try to obtain dynamically the IP but that does not work. I forgot to remove it later. I will remote it now.